src/libraries/System.Security.Cryptography.Cose/tests/CoseTestHelpers.cs

```diff
@@ -469,24 +604,57 @@ internal static Stream GetTestStream(byte[] content, StreamKind streamKind = Str
             return ms;
         }
 
-        internal static CoseSigner GetCoseSigner(AsymmetricAlgorithm key, HashAlgorithmName hash, CoseHeaderMap? protectedHeaders = null, CoseHeaderMap? unprotectedHeaders = null, RSASignaturePadding? padding = null)
+        internal static CoseSigner GetCoseSigner(IDisposable key, HashAlgorithmName? hash, CoseHeaderMap? protectedHeaders = null, CoseHeaderMap? unprotectedHeaders = null, RSASignaturePadding? padding = null)
         {
             if (key is RSA rsa)
             {
-                return new CoseSigner(rsa, padding ?? RSASignaturePadding.Pss, hash, protectedHeaders, unprotectedHeaders);
+                Assert.NotNull(hash);
```

Comment:
```suggestion
                AssertExtensions.TrueExpressino(hash.HasValue);
```

Would be a more informative failure.